pandas/core/internals.py

```diff
@@ -4708,15 +4709,20 @@ def _safe_reshape(arr, new_shape):
     return arr
 
 
-def _transform_index(index, func):
+def _transform_index(index, func, level=None):
     """
     Apply function to all values found in index.
 
     This includes transforming multiindex entries separately.
+    Only apply function to one level of the MultiIndex if level is specified.
 
     """
     if isinstance(index, MultiIndex):
-        items = [tuple(func(y) for y in x) for x in index]
+        if level is not None:
+            items = [tuple(func(y) if i == level else y
```

Comment:
Well, that's how rename is implemented at the moment .. 
And AFAIK there is not a rename method on Index itself? (there is, but that is to change the name attribute, so something else)